### DISCUSSION TITLE
Send PDF output directly to printer without user interaction

### DISCUSSION THREAD

**Discussion by maxbobse** [eclipse-birt/birt#1580] [Q&A] [answered]
Is it possible to render a report as PDF and send it straight to a pre-defined (or default) printer to print? Ideally the usual print dialog does not pop up and the report gets printed right away.

This question is not about if its possible to render a report as PDF, that part I know :)
It is about sending it automatically to a printer afterwards.

I am using BIRT 4.7

**Comment by hvbtup** [comment, marked as answer]
No, not with BIRT.

But you can integrate a 3rdParty library into your application in such a way that BIRT generates a PDF file in a temporary directory and the 3rdParty library is responsible for printing it. 

However, this not "plug&play". The point is that you have to integrate BIRT and the PDF printing library into your application.
For example, you cannot use the WebViewer, at least not as-is.

In our commercial application we are using a `RunAndRenderTask` to create a PDF file with BIRT, then we are using a [commercial library](https://www.pdf-tools.com/products/viewing-printing/pdf-printer/) for printing on Windows. 

On Linux, of course, printing work very differently. It's usually easy to print a PDF file, but the print quality might not be as perfect as with Windows.

**Reply by maxbobse** [reply]
Thanks for the info!
